src/main/java/edu/ucsb/cs156/organic/controllers/SchoolController.java

```diff
@@ -112,34 +111,21 @@ public Object deleteSchool(
   @PreAuthorize("hasRole('ROLE_ADMIN')")
     @PostMapping("/post")
     public School postSchool(
-        @Parameter(name="abbrev", description="university domain name", example="ucsb") @RequestParam String abbrev,
-        @Parameter(name="name", description="University name") @RequestParam String name,
-        @Parameter(name="termRegex", description="Format: Example [WSMF]\\d\\d") @RequestParam String termRegex,
-        @Parameter(name="termDescription", description="Enter quarter, e.g. F23, W24, S24, M24") @RequestParam String termDescription,
-        @Parameter(name="termError", description="input error?") @RequestParam String termError)
-        {
-
-        School school = School.builder().build();
-        school.setAbbrev(abbrev);
-        school.setName(name);
-        school.setTermRegex(termRegex);
-        school.setTermDescription(termDescription);
-        school.setTermError(termError);
-
-        if (!termDescription.matches(school.getTermRegex())) {
-            throw new IllegalArgumentException("Invalid termDescription format. It must follow the pattern " + school.getTermRegex());
-        }
+        @Parameter(name = "school", description="school in json format") @RequestBody School school
+        )
+    {
 
-        if (!abbrev.equals(abbrev.toLowerCase())){
+        if (!school.getAbbrev().equals(school.getAbbrev().toLowerCase())){
             throw new IllegalArgumentException("Invalid abbrev format. Abbrev must be all lowercase");
         }
 
+        if (!school.getTermDescription().matches(school.getTermRegex())) {
+            throw new IllegalArgumentException("Invalid termDescription format. It must follow the pattern " + school.getTermRegex());
+        }
+
```

Comment:
As long as we are editing this file, let's fix this, because this was never the intention of this field.

The term description is just supposed to be the word "Quarter" (e.g. for UCSB) or "Semester" (e.g. for Washington State).

Just remove this.

```suggestion
```